R/uniDimBayesianCalcCoeff.R

```diff
@@ -0,0 +1,484 @@
+  }
+  return(out)
+}
+
+.BayesianStdDev <- function(jaspResults, dataset, options, model) {
+  if (!is.null(.getStateContainerB(jaspResults)[["sdObj"]]$object))
+    return(.getStateContainerB(jaspResults)[["sdObj"]]$object)
+
+  out <- model[["sd"]]
+  if (is.null(out))
+    out <- list()
+  if (options[["sdScale"]] && is.null(model[["empty"]])) {
+    if (options[["sdMethod"]] == "sumScores")
+      out[["est"]] <- sd(rowSums(dataset, na.rm = T))
+    else
+      out[["est"]] <- sd(rowMeans(dataset, na.rm = T))
+
+    out[["cred"]] <- c(NA_real_, NA_real_)
+    stateContainer <- .getStateContainerB(jaspResults)
+    stateContainer[["sdObj"]] <- createJaspState(out, dependencies = c("sdScale", "sdMethod"))
+  }
+  return(out)
+}
+
+
+
+.BayesianItemRestCor <- function(jaspResults, dataset, options, model) {
+  if (!is.null(.getStateContainerB(jaspResults)[["itemRestObj"]]$object))
+    return(.getStateContainerB(jaspResults)[["itemRestObj"]]$object)
+
+  out <- model[["itemRestCor"]]
+  if (is.null(out))
+    out <- list()
+  # is box even checked?
+  if (options[["itemRestCor"]] && is.null(model[["empty"]])) {
+    ciValueItem <- options[["credibleIntervalValueItem"]]
+
+    if (is.null(out[["itemSamp"]])) {
+      startProgressbar(options[["noSamples"]] * options[["noChains"]] * ncol(dataset))
+      if (anyNA(dataset) && !model[["pairwise"]]) { # item rest cor needs its own missing handling, at least for listwise
+        pos <- which(is.na(dataset), arr.ind = TRUE)[, 1]
+        dataset <- dataset[-pos, ]
+      }
+      # dataset <- scale(dataset, scale = F)
+      if (options[["setSeed"]])
+        set.seed(options[["seedValue"]])
+      out[["itemSamp"]] <- .itemRestCor(dataset, options[["noSamples"]], options[["noBurnin"]],
+                              options[["noThin"]], options[["noChains"]], model[["pairwise"]],
+                              callback = progressbarTick)
+    }
+
+    out[["itemEst"]] <- apply(out[["itemSamp"]], 3, mean)
+    out[["itemCred"]] <- coda::HPDinterval(coda::mcmc(apply(out[["itemSamp"]], 3, as.vector)),
+                                           prob = ciValueItem)
+
+    stateContainer <- .getStateContainerB(jaspResults)
+    stateContainer[["itemRestObj"]] <- createJaspState(out,
+                                                       dependencies = c("itemRestCor", "credibleIntervalValueItem"))
+  }
+
+  return(out)
+}
+
+.BayesianItemMean <- function(jaspResults, dataset, options, model) {
+  if (!is.null(.getStateContainerB(jaspResults)[["itemMeanObj"]]$object))
+    return(.getStateContainerB(jaspResults)[["itemMeanObj"]]$object)
+
+  out <- model[["itemMean"]]
+  if (is.null(out))
+    out <- list()
+  # is box even checked?
+  if (options[["itemMean"]] && is.null(model[["empty"]])) {
+    out[["itemEst"]] <- numeric(ncol(dataset))
+    for (i in 1:ncol(dataset)) {
+      out[["itemEst"]][i] <- mean(dataset[, i], na.rm = T)
```

Comment:
can't this be `colMeans(dataset, na.rm = TRUE)`? Also, please use `TRUE` and `FALSE` over `T` and `F`.